eclass/nodejs-mod.eclass

```diff
@@ -0,0 +1,110 @@
+# Copyright 2019-2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+# @ECLASS: nodejs-mod.eclass
+# @MAINTAINER:
+# Fco. Javier Félix <web@inode64.com>
+# @AUTHOR:
+# Fco. Javier Félix <web@inode64.com>
+# @SUPPORTED_EAPIS: 7 8
+# @BLURB: An eclass for build NodeJS projects
+# @DESCRIPTION:
+# An eclass providing functions to build NodeJS projects
+#
+# Changelog:
+#   Initial version from:
+#       https://github.com/gentoo/gentoo/pull/930/files
+#       https://github.com/samuelbernardo/ssnb-overlay/blob/master/eclass/npm.eclass
+#       https://github.com/gentoo-mirror/lanodanOverlay/blob/master/eclass/nodejs.eclass
+#       https://github.com/Tatsh/tatsh-overlay/blob/master/eclass/yarn.eclass
+
+#
+# Build package for node_modules:
+#   npm:
+#   npm install --audit false --color false --foreground-scripts --progress false --verbose --ignore-scripts
+#
+#   yarn:
+#   yarn install --color false --foreground-scripts --progress false --verbose --ignore-scripts
+#
+#   Create archive in tar:
+#   tar --create --auto-compress --file foo-1-node_modules.tar.xz foo-1/node_modules/
+
+case ${EAPI} in
+7 | 8) ;;
+*) die "${ECLASS}: EAPI ${EAPI:-0} not supported" ;;
+esac
+
+if [[ -z ${_NODEJS_MOD_ECLASS} ]]; then
+_NODEJS_MOD_ECLASS=1
+
+inherit nodejs
+
+if has nodejs-pack ${INHERITED}; then
+    eerror "nodejs-mod and nodejs-pack eclass are incompatible"
+fi
+
+RDEPEND+=" net-libs/nodejs:="
+
+# @FUNCTION: nodejs-mod_src_prepare
+# @DESCRIPTION:
+# Nodejs preparation phase
+nodejs-mod_src_prepare() {
+    debug-print-function "${FUNCNAME}" "${@}"
+
+    if [[ ! -e package.json ]]; then
+        eerror "Unable to locate package.json"
+        eerror "Consider not inheriting the nodejs eclass."
+        die "FATAL: Unable to find package.json"
+    fi
+
+    default_src_prepare
+}
+
+# @FUNCTION: nodejs-mod_src_compile
+# @DESCRIPTION:
+# General function for compiling a nodejs module
+nodejs-mod_src_compile() {
+    debug-print-function "${FUNCNAME}" "${@}"
+
+    if [[ -d node_modules ]]; then
+        einfo "Compile native addon modules"
+        find node_modules/ -name binding.gyp -exec dirname {} \; | while read -r dir; do
+            pushd "${dir}" >/dev/null || die
+            # shellcheck disable=SC2046
```

Comment:
I think we do not "canonize" the use of shellcheck in ebuilds/eclasses, and so those comments to make spellcheck happy are just noise.

Reply:
As in go-module.eclass they are also used, I put them

Reply:
Please don't for now, until shellcheck supports ebuilds properly.

Reply:
I like this change from mine as I did not really look into how to properly determine things to rebuild.